### DISCUSSION TITLE
Specify env variable name in nested settings class

### DISCUSSION THREAD

**Discussion by palle-k** [pydantic/pydantic#8221] [Question]
I would like to specify the name of an environment variable, from which a field in a nested model is loaded.

Example:

```python
from pydantic import BaseModel
from pydantic_settings import BaseSettings, SettingsConfigDict

class NestedModel(BaseModel):
    foo: str = Field(...)  # this should be loaded from SOME_ENV_VAR
    bar: str  # this should just be loaded from NESTED__BAR

class MySettings(BaseSettings):
    nested: NestedModel

    model_config = SettingsConfigDict(env_nested_delimiter="__")
```

With Pydantic 2.0, using `Field(env="SOME_ENV_VAR")` no longer works. However, when I use the methods described in the docs, `validation_alias` or `alias`, the prefix from `MySettings` is already applied, meaning that I can only access env variables that have a `NESTED__` prefix.

How can I circumvent this behaviour and parse a value from an environment variable given it's exact name?

I would like to avoid going with `os.getenv` as the default factory, because I would like to keep using Pydantic for loading my env variables from a variety of sources (including some custom sources, .env files, regular env vars)

**Comment by hramezani** [comment]
Yeah, `env` is removed in `pydantic-settings` and replaced by `validation_alias`.

I can't see any easy way other than the thing you mentioned to not consider `NESTED__` prefix when loading env variable for `foo`.

Probably, You can use an override version of the [EnvSettingsSource](https://github.com/pydantic/pydantic-settings/blob/15ef2a9c947d011c16a40ef115bf90fcceaebd1d/pydantic_settings/sources.py#L359). Take a look at [customise settings sources doc](https://docs.pydantic.dev/latest/concepts/pydantic_settings/#customise-settings-sources), it might help.

**Comment by palle-k** [comment]
Thanks for your input, @hramezani! I was able to get this to work in a very hacky way by injecting all env variables into each nesting level of the exploded env vars:

```python
class NestedFallbackDotEnvSettingsSource(FrameworkSettingsDotEnvSource):
    def explode_env_vars(self, field_name: str, field: FieldInfo, env_vars: Mapping[str, str | None]) -> dict[str, Any]:
        def deep_inject(target: dict[str, Any], source: Mapping[str, Any]) -> dict[str, Any]:
            updated_target = {
                key: deep_inject(value, source) if isinstance(value, dict) else value for key, value in target.items()
            }
            for key, value in source.items():
                if key not in updated_target:
                    updated_target[key] = value

            return updated_target

        parent_results = super().explode_env_vars(field_name, field, env_vars)
        updated_results = deep_inject(parent_results, env_vars)
        return updated_results
```

However, I do not approve of this solution. Something that previously was one property of `Field` now takes a complicated workaround that pollutes the exploded env variable dictionaries and potentially leads to incorrect parses of env variables.

**Comment by palle-k** [comment]
Another workaround that I found is to have the nested model inherit from `BaseSettings` as well. But that also requires me do copy `model_config` from the containing BaseSettings.
